### ISSUE TITLE
VMware Tools not installed properly in Catalina box

### ISSUE THREAD

**Issue by bacongravy** [bacongravy/macinbox#45]
Running `macinbox` on Catalina successfully creates a Vagrant box in `vmware_desktop` format but when the box boots up the OS reports an error loading the VMware Tools kernel extension. Besides the error message, the failure to load the kernel extension results in the screen resolution being stuck at 1024x768, and shared folders not working.

Using VMware Fusion's menu command *Virtual Machine > Reinstall VMware Tools* after the box boots successfully resolves the issue.

**Comment by StefanScherer** [comment]
I just learned about this tool. Thank you @bacongravy for the hard work. I really appreciate that there is a tool to create newer macOS Vagrant boxes.

I just built a Mojave 10.14.6 box and VMware tools also aren't installed correctly. This message appears in the VM after a `vagrant up`:

<img width="635" alt="Bildschirmfoto 2019-11-13 um 12 19 33" src="https://user-images.githubusercontent.com/207759/68759388-dbcad280-060f-11ea-9694-38d1a49c1151.png">


On the host `vagrant up` also "hangs":
```
$ vagrant up
Bringing machine 'default' up with 'vmware_fusion' provider...
==> default: Cloning VMware VM: 'macinbox-10.14.6'. This can take some time...
==> default: Verifying vmnet devices are healthy...
==> default: Preparing network adapters...
==> default: Starting the VMware VM...
==> default: Waiting for the VM to receive an address...
```

My setup is Vagrant 2.2.5, macOS 10.14.6, VMware Fusion Pro 11.5.1, macinbox 4.0.0

**Comment by bacongravy** [comment]
Hi @StefanScherer thank you for the feedback!  I'm sorry the tool isn't working properly for you. I have successfully created a box from a macOS 10.14.6 installer, but it was with a previous version of VMware Fusion Pro; maybe something about the Tools package has changed.  You could try using VMware Fusion Pro 11.1.1 to see if that makes a difference.  I'd also be curious to know if manually installing the tools after the box has booted resolves the issue for you.

I'll see if I can replicate the failure you are seeing. I don't currently have a Mojave system configured to test this with so it might not happen immediately.

**Comment by StefanScherer** [comment]
Yes, I could install VMware tools manually again and restarted. Then the box works fine. I tar‘ed that to a new box file and tested it on a second machine. 

I‘ll keep my one MBP on Mojave for a while (until the updater tricks me when I click the wrong button in a hurry).

I‘ve update Fusion just today. But, when I find some time I can downgrade to do some testing. 

It‘s always this magic triangle macOS version/Vagrant version/Fusion version that you have to keep in mind.

**Comment by egandro** [comment]
I know about this "bug" but was to lazy to fix it:

Check this:

- https://github.com/bacongravy/macinbox/blob/46939cda43fbf7902759bf46cc8ee7426e608dd0/lib/macinbox/actions/create_vmdk_from_image.rb

We do some inserts to a mysql DB in order to whitelist the kext a priori to the native boot (during the installation).

This needs to be enhanced for Vmware Fusion 11.5 (it's only done for 11.0).

So my idea - just add VBox 6.x and 11.5 in the .rb file.

Sorry again - I knew all about this but I was to lazy to do it.

**Comment by bacongravy** [comment]
@egandro Thanks for the tip.  I'll take a look at what the new values should be, unless you happen to know them offhand?  And the values for VBox?

**Comment by Footpad** [comment]
@bacongravy I created a PR here that allows VMware Tools for Fusion 11.5.3 (not sure about other version support) to load successfully: https://github.com/bacongravy/macinbox/pull/56

**Comment by bacongravy** [comment]
@Footpad Thank you! I'll take a look now.

**Comment by bacongravy** [comment]
I have verified that the PR provided by @Footpad resolves this issue.

**Comment by bacongravy** [comment]
This will be fixed in the v4.1.0 release.

**Comment by bacongravy** [comment]
This has been released. Closing.
